### PR TITLE
Gauge percentage color fix w/test

### PR DESCRIPTION
**Pull request by go-hacks**
I figured out how to make a new request so here's a proper one you can easily merge. I put the test in _test where it should be and gauge.go only has the one line change.

This is the issue this addresses btw.

https://github.com/gizak/termui/issues/306

The first guys edit doesn't quite get us what we want albeit better than the original. Copying the base color as mine does works much better than even the originally intended concept that simply doesn't work anyways.

### CODE REVIEW

**Comment by dcorbe** [comment]
@go-hacks Can you post a code snippet demonstrating the problem which led to this PR?  If you run make in the top-level directory, the gauge tests fail.  I'd prefer to address whatever is wrong with ModifierReverse.

**Comment by go-hacks** [comment]
I put the link to the discussion of the problem above. Someone else showed it quite nicely there. My fix is basically the same thing but there's no need to call the NewStyle when we just need the barcolor even though both options give the desired result.

If you look at that #306 issue, it shows all this. When the gauge hits the percentage it tends to blank it out and you can't see it. Admittedly with this fix, like the other guy said too, if you choose 2 colors that are the same for the bar and the percentage label then you can't see the numbers as the bar crosses it but honestly that's how it should be. You should be able to set whatever color you want for the bar and the percentage and not have it change to some "magical reverse" color. (which doesn't work anyways). Then you'd have to hope it chooses a color you even want when it passes over. 

You're more than welcome to try and play with the modifier reverse but I'll tell you right now it won't work. I tried every bit level modification possible with that concept and none give the desired result. I also stated this in that #306 issue. 

Whatever the original creator was trying for with that modifier simply didn't work. You can't just flip a single bit like that and get the desired result. I tried all possible values already. No matter what bit you flip the color comes out wrong. There's a couple that are close but still wrong.

As for the test, it's self contained. You would just build it by itself. I guess i've never tried to build this project on it's own from the root dir like that. I only import it into my projects and build them. If it causes build issues, it can be removed. It was just to prove the concept worked. It doesn't need to stay there. It merely cycles through the colors and runs the gauge bar through the percentage so you can see it works properly. (it doesn't blank the percentage but if you choose the same color for the bar and percentage then it's not readable, naturally, is it should be)

The fix is really just that one line. You wouldn't even need to commit this request to change it quick.

**Comment by dcorbe** [comment]
> I'm not even sure why you would...Just to build the demo or something?

A test build is almost always necessary for CI/CD pipelines.  I plan on setting up Travis for this project.  But I do get your point.  I'll meditate on this a bit and see if I can fix the demos without ModifierReverse.  

Thank you for your patience and always I do appreciate your contributions.

**Comment by go-hacks** [comment]
Alright cool. Good to know. Thanks for info. I'm still learning some of this stuff too so I'm glad we can help each other. :) This situation is definitely a weird one. I get it. I spent many hours pouring over it as well. I think the original creator had an interesting idea but it didn't pan out as intended so we just gta make the best of it. At least it's not a large edit so if the original creator comes back and finds the solution to what he was after it won't be a big deal at all to change it. If we could actually talk to him, we could know more but we haven't been able to, hence the fork lol.

**Comment by go-hacks** [comment]
I just noticed something interesting... So, if the gauge is used inside the termui directory (I put my test in the _examples folder and updated the import paths as needed) it behaves differently than importing it into another project outside of the termui directory. If I build the test elsewhere (with the original code) it behaves like the issue shows with the blanking... However, if that same test is built inside the _examples folder, it doesn't blank out the percentage but it makes it black. Regardless of bar color it changes the text to black. 

For some reason that section of code behaves differently if imported from outside of the project. I don't know enough to understand why that would be but it is...So knowing this, there definitely could be some other issue that I wouldn't know where to begin with. However, the proposed fix is still better than simply turning the text black regardless of the color for the reasons previously discussed. Even if you figured out how to get the modifier functionality to work outside of the termui folder.

I just wanted to point this out as it implies there's some other problem with that modifier function that's not specifically with the ModiferReverse but how it's applying it which is odd because the other modifiers seem to work as intended.

File: _test/gaugeTest.go [hunk 1]
```diff
@@ -0,0 +1,60 @@
+package main
+
+import (
+	"log"
+	"time"
+	ui "github.com/gizak/termui"
```
Comment (go-hacks) [inline _test/gaugeTest.go:6]: I forgot to change these to your dcorbes. If those are changed and we move the test to _examples, then the make from the main dir doesn't complain :)

Comment (dcorbe) [reply _test/gaugeTest.go:6]: @go-hacks I've fixed the absolute import paths in this commit: https://github.com/dcorbe/termui-dpc/commit/3a48ba1f16a13e14308340991a99251c5fde088a

If you do a git pull from your local copy you should have a buildable copy.

Comment (go-hacks) [reply _test/gaugeTest.go:6]: awesome! thanks!

**Pull request merged**